lib/plugs/authentication_plug.ex

```diff
@@ -0,0 +1,19 @@
+defmodule Shareguru.AuthenticationPlug do
+  import Plug.Conn
+  import Plug.Session
+
+  def init(options), do: options
+
+  def call(conn, options) do
+    case get_session(conn, :current_user) do
+      nil -> do_redirect(conn, Shareguru.Router.Helpers.auth_path(Shareguru.Endpoint, :request, "google"))
+      _ -> conn
```

Comment:
I've a feeling this could be also written using pattern matching (with 2 function definitions), but it might not be worth muling over.